vsgenstubs4/_vapoursynth.part.pyi

```diff
@@ -426,83 +653,183 @@ class ColorPrimaries(enum.IntEnum):
     PRIMARIES_EBU3213_E: 'ColorPrimaries'
 
 
-PRIMARIES_BT709: ColorPrimaries
-PRIMARIES_UNSPECIFIED: ColorPrimaries
-PRIMARIES_BT470_M: ColorPrimaries
-PRIMARIES_BT470_BG: ColorPrimaries
-PRIMARIES_ST170_M: ColorPrimaries
-PRIMARIES_ST240_M: ColorPrimaries
-PRIMARIES_FILM: ColorPrimaries
-PRIMARIES_BT2020: ColorPrimaries
-PRIMARIES_ST428: ColorPrimaries
-PRIMARIES_ST431_2: ColorPrimaries
-PRIMARIES_ST432_1: ColorPrimaries
-PRIMARIES_EBU3213_E: ColorPrimaries
+PRIMARIES_BT709: Literal[ColorPrimaries.PRIMARIES_BT709]
+PRIMARIES_UNSPECIFIED: Literal[ColorPrimaries.PRIMARIES_UNSPECIFIED]
+PRIMARIES_BT470_M: Literal[ColorPrimaries.PRIMARIES_BT470_M]
+PRIMARIES_BT470_BG: Literal[ColorPrimaries.PRIMARIES_BT470_BG]
+PRIMARIES_ST170_M: Literal[ColorPrimaries.PRIMARIES_ST170_M]
+PRIMARIES_ST240_M: Literal[ColorPrimaries.PRIMARIES_ST240_M]
+PRIMARIES_FILM: Literal[ColorPrimaries.PRIMARIES_FILM]
+PRIMARIES_BT2020: Literal[ColorPrimaries.PRIMARIES_BT2020]
+PRIMARIES_ST428: Literal[ColorPrimaries.PRIMARIES_ST428]
+PRIMARIES_ST431_2: Literal[ColorPrimaries.PRIMARIES_ST431_2]
+PRIMARIES_ST432_1: Literal[ColorPrimaries.PRIMARIES_ST432_1]
+PRIMARIES_EBU3213_E: Literal[ColorPrimaries.PRIMARIES_EBU3213_E]
 
 
 ###
 # VapourSynth Environment SubSystem
+
+
 class EnvironmentData:
-    """
-    Contains the data VapourSynth stores for a specific environment.
-    """
+    def __init__(self) -> NoReturn: ...  # type: ignore[misc]
+
+
+class EnvironmentPolicy:
+    def on_policy_registered(self, special_api: 'EnvironmentPolicyAPI') -> None: ...
+
+    def on_policy_cleared(self) -> None: ...
+
+    @abstractmethod
+    def get_current_environment(self) -> Union[EnvironmentData, None]: ...
+
+    @abstractmethod
+    def set_environment(self, environment: Union[EnvironmentData, None]) -> Union[EnvironmentData, None]: ...
+
+    def is_alive(self, environment: EnvironmentData) -> bool: ...
+
+
+class StandaloneEnvironmentPolicy(EnvironmentPolicy):
```

Comment:
This is internal VapourSynth api. You will **never** get an instance of this class in your code, and you can't instantiate it. Please hide this from the .pyi.